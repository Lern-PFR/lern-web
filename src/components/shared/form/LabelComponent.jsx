import PropTypes from 'prop-types';
import { labelStyle } from 'theme/formStyles';
import DynamicTextComponent from 'components/shared/typography/DynamicTextComponent';
import { canon, greatPrimer, doublePica, bodyCopy, brevier, longPrimer, minion, paragon, pica, trafalgar } from 'theme/textStyles';

/**
 * @name LabelComponent
 * @description A component used to display styled label element.
 *
 * @author Christopher Walker
 *
 * @param {string}	[element]		: The id of the element that the label is associated to.
 * @param {string}	[typography]	: The text style of the label text.
 * @param {string}	children		: The test to be displayed in the label.
 */
const LabelComponent = ({ children, typography, forId, ...otherProps }) => (
	<DynamicTextComponent
		{...(() => {
			switch (typography) {
				case 'canon':		return canon;
				case 'trafalgar':	return trafalgar;
				case 'paragon':		return paragon;
				case 'greatprimer':	return greatPrimer;
				case 'doublepica':	return doublePica;
				case 'bodycopy':	return bodyCopy;
				case 'pica':		return pica;
				case 'longprimer':	return longPrimer;
				case 'minion':		return minion;
				default:			return brevier;
			}
		})()}
		tag="label"
		{...labelStyle}
		htmlFor={forId}
		{...otherProps}
	>
		{children}
	</DynamicTextComponent>
);

LabelComponent.propTypes = {
	children: PropTypes.string.isRequired,
	typography: PropTypes.string,
	forId: PropTypes.string,
};

LabelComponent.defaultProps = {
	typography: 'brevier',
	forId: '',
};

export default LabelComponent;
