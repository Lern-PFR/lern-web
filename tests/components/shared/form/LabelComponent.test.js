import { shallow } from 'enzyme';
import LabelComponent from 'components/shared/form/LabelComponent';
import { cleanup } from '@testing-library/react';

describe('LabelComponent', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match previous snapshot', () => {
		const sut = (<LabelComponent>Test label</LabelComponent>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
