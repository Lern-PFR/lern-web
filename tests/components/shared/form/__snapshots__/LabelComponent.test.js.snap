// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelComponent should match previous snapshot 1`] = `
<DynamicTextComponent
  color="#241C15"
  cursor="pointer"
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize=".778rem"
  fontWeight={400}
  htmlFor=""
>
  Test label
</DynamicTextComponent>
`;

exports[`LabelComponent should match previous snapshot with custom typography 1`] = `
<DynamicTextComponent
  color="#241C15"
  cursor="pointer"
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize=".778rem"
  fontWeight={400}
  htmlFor=""
  letterSpacing={0.32}
  lineHeight="1.125em"
  tag="label"
  typography="doublepica"
>
  Test label
</DynamicTextComponent>
`;
